analyzer/vulnerability-consumer/src/main/java/eu/fasten/analyzer/vulnerabilityconsumer/db/MetadataUtility.java

```diff
@@ -0,0 +1,375 @@
+                .and(Callables.CALLABLES.LINE_START.le(lineNumber))
+                .and(Callables.CALLABLES.LINE_END.ge(lineNumber))
+                .fetch();
+
+        return fasten_uris.stream().map(Record1::component1).collect(Collectors.toList());
+    }
+
+    /**
+     * Retrieves the package_version_id given the purl of the package version.
+     *
+     * @param purl - follows purl specifications
+     * @return negative if it cannot be found
+     */
+    public Long getPackageVersionId(String purl, DSLContext context, Long pkgId) {
+        logger.info("Looking for package_version_id of " + purl);
+        PURLPackage packageObject = PURLPackage.getObjectFromPurl(purl);
+
+        assert pkgId > 0L;
+
+        var pkgVersionRecord = context.select(PackageVersions.PACKAGE_VERSIONS.ID)
+                .from(PackageVersions.PACKAGE_VERSIONS)
+                .where(PackageVersions.PACKAGE_VERSIONS.PACKAGE_ID.equal(pkgId))
+                .and(PackageVersions.PACKAGE_VERSIONS.VERSION.equal(packageObject.getVersion()))
+                .fetchOne();
+        return pkgVersionRecord != null ? pkgVersionRecord.component1() : -1L;
+    }
+
+    /**
+     * Retrieve the fileId of the file that was patched.
+     *
+     * @param packageVersionId - Long pkg version ID
+     * @param filepath         - path to the file
+     * @return -1 if the file cannot be found
+     */
+    public Long getFileId(Long packageVersionId, String filepath, DSLContext context) {
+        var fileRecords = context.select(Files.FILES.ID, Files.FILES.PATH)
+                .from(Files.FILES)
+                .where(Files.FILES.PACKAGE_VERSION_ID.equal(packageVersionId))
+                .fetch();
+        var file = fileRecords.stream()
+                .filter(fr -> (!fr.component2().equals("") && filepath.contains(fr.component2())))
+                .map(Record2::component1).collect(Collectors.toList());
+        return file.size() > 0 ? file.get(0) : -1L;
+    }
+
+    /**
+     * Gets the moduleId that corresponds to the file.
+     *
+     * @param fileId - Long fileId
+     * @return list of module Ids
+     */
+    public List<Long> getModuleIds(Long fileId, DSLContext context) {
+        var moduleRecords = context.select(ModuleContents.MODULE_CONTENTS.MODULE_ID)
+                .from(ModuleContents.MODULE_CONTENTS)
+                .where(ModuleContents.MODULE_CONTENTS.FILE_ID.equal(fileId))
+                .fetch();
+        return moduleRecords.stream().map(Record1::component1).collect(Collectors.toList());
+    }
+
+    /**
+     * Returns package ID given PURL and CONTEXT (forge).
+     *
+     * @param purl    - PURL string
+     * @param context - DSLContext
+     * @return -1L if not existing
+     */
+    public Long getPackageId(String purl, DSLContext context) {
+        var coordinate = purl.substring(purl.indexOf("/") + 1)
+                .split("@")[0]
+                .replace("/", ":");
```

Comment:
Intead of `":"` you could use `Constants.mvnCoordinateSeparator`.
P.S. It is related only for Maven artifacts, so make sure you accommodate other languages/forges (PyPI and Debian), but that's for later.

Reply:
Good catch. I have written a new method to find the appropriate separator to use. I will uncomment the lines for Debian and PyPi once the `Constants` are added to `core`